cpp/src/arrow/dataset/partition.cc

```diff
@@ -827,7 +827,7 @@ class HivePartitioningFactory : public KeyValuePartitioningFactory {
   Result<std::shared_ptr<Partitioning>> Finish(
       const std::shared_ptr<Schema>& schema) const override {
     if (dictionaries_.empty()) {
-      return std::make_shared<HivePartitioning>(schema, dictionaries_);
+      return std::make_shared<HivePartitioning>(arrow::schema({}), dictionaries_);
```

Comment:
Nit: Is the `arrow::` prefix necessary here?

Reply:
There is an arugment `schema` to the function, so it would need either refactoring argument or using the namespace. Any thoughts?

Reply:
Let's disambiguate then.  Can you change the argument name to `dataset_schema`?

Reply:
Sure. I will.